src/logic/safe/store/reducer/pendingTransactions.ts

```diff
@@ -0,0 +1,62 @@
+import { Action, handleActions } from 'redux-actions'
+import { Set } from 'immutable'
+
+import session from 'src/utils/storage/session'
+import { PENDING_TRANSACTIONS_ACTIONS } from 'src/logic/safe/store/actions/pendingTransactions'
+import { ChainId } from 'src/config/chain.d'
+import { _getChainId } from 'src/config'
+
+export const PENDING_TRANSACTIONS_ID = 'pendingTransactions'
+
+type SafeTxHash = string
+export type PendingTransactionsState = Record<ChainId, Set<SafeTxHash>>
+
+// It is necessary to convert session-stored state back to a Set otherwise it is an array
```

Comment:
I would just use a plain JS object given all the hassle.
`{ '4': { '0x44364y': true } }`

Reply:
I guess I'll revert all the complex logic then.